### ISSUE TITLE
What is the best way to have different transitions based on the app state?

### ISSUE THREAD

**Issue by tom-sherman** [illinois/next-page-transitions#15]
Say that I have two pages `PageA` and `PageB`.

I want to transition between `PageA` -> `PageB` using `TransitionX`.

But when I'm navigating between `PageB` -> `PageA` I want to use `TransitionY`.

Is this easily achieved with this package or is it best to use a more custom solution? (React Spring or similar)

**Comment by nwalters512** [comment]
I don't think there's a particularly easy way to do that with this package. It'd be worth looking into supporting this though. Do you have an idea of what an API for this might look like?

**Comment by tky5622** [comment]
@tom-sherman 
I think It would be possible by conditionally adding styled jsx with routeChangeStart or something.
 
Anyway, I want api  looks like
onEnter : { pageName: css }
onLeave : { pageName: css } 

thanks.

**Comment by kapoko** [comment]
I just found a solution for this. In my case I had pages that were dark themed and some that were light themed so I wanted to make the transition aware of the next page. I use pageProps for this. Here's the simplified code:

```
// pages/index.js

const Index = props => (
    <h1>Index page</h1>
)

Index.getInitialProps = async function () {
    return { theme: 'light' }
}

export default Index
```

```
// pages/about.js

const About = props => (
    <h1>About page</h1>
)

Index.getInitialProps = async function () {
    return { theme: 'dark' }
}

export default About
```

Then in `_app.js` we can fetch the `theme` prop and pass it to the `PageTransition`'s component `classNames` prop.

```
// pages/_app.js

class MyApp extends App {
    static async getInitialProps({ Component, router, ctx }) {
        let pageProps = {};
        
        if (Component.getInitialProps) {
            pageProps = await Component.getInitialProps(ctx);
        }
        
        return { pageProps };
    }

    render() {
        const { Component, pageProps } = this.props;

        return (
            <Container>
                <PageTransition timeout={1000} classNames={`transition-${pageProps.theme}`}>
                   <Component {...pageProps} />
                </PageTransition>
            </Container>
        )
    }
}
export default MyApp
```

Now I can use the classes `transition-light-enter` `transition-light-enter-active` `transition-light-enter-done` `transition-dark-enter` `transition-dark-enter-active` `transition-dark-enter-done` etc. etc. to style the different transitions.

**Comment by nwalters512** [comment]
@tom-sherman can you adopt the above to do what you're looking for?

**Comment by tom-sherman** [comment]
@nwalters512 absolutely, great solution. May I suggest adding it as an example in the docs?

**Comment by kapoko** [comment]
Simplified the code a bit as I learned `getInitialProps` in `_app.js` wasn't necessary.

**Comment by nwalters512** [comment]
@tom-sherman sure, I'd be happy to.

@kapoko would you be willing to build a small self-contained example for the `examples` directory? Combined with a description in the Readme, that'd be really useful. If you'd want to also submit a PR adding this to the docs, that would be great 🙏

**Comment by kapoko** [comment]
Sure thing.

https://github.com/illinois/next-page-transitions/pull/39

**Comment by nwalters512** [comment]
I just merged #39, going to close this out since we now have an example of how to do this.
